lib/sorbet-rails/model_plugins/active_record_named_scope.rb

```diff
@@ -7,15 +7,23 @@ class SorbetRails::ModelPlugins::ActiveRecordNamedScope < SorbetRails::ModelPlug
   def generate(root)
     model_class_rbi = root.create_class(self.model_class_name)
 
-    @model_class.methods.sort.each do |method_name|
+    # Named scope methods are dynamically defined by the `scope` method so their
+    # source_location is `lib/active_record/scoping/named.rb`. So we find scopes
+    # by two criteria:
+    # - they are defined in 'activerecord/lib/active_record/scoping/named.rb'
+    # - they are not one of the methods actually defined in that file's source.
+    #
+    # These methods are defined within `lib/active_record/scoping/named.rb` and should
+    # be ignored for the purposes of this plugin.
+    # See: https://github.com/rails/rails/blob/master/activerecord/lib/active_record/scoping/named.rb
+    non_dynamic_methods = %i(all scope_for_association default_scoped default_extensions scope valid_scope_name?)
```

Comment:
I'm thinking how to do this more systematically, and realize we can use "instance_methods" to find these methods, eg:
```
irb(main):023:0> pp ActiveRecord::Scoping::Named::ClassMethods.instance_methods.sort
[:all, :default_extensions, :default_scoped, :scope, :scope_for_association]
=> [:all, :default_extensions, :default_scoped, :scope, :scope_for_association]
irb(main):024:0> pp ActiveRecord::Scoping::Named::ClassMethods.private_instance_methods.sort
[:valid_scope_name?]
=> [:valid_scope_name?]
irb(main):025:0> pp ActiveRecord::Scoping::Named::ClassMethods.protected_instance_methods.sort
[]
=> []
```

I was also thinking we could just exclude all methods defined in ActiveRecord::Base, but someone may try to define a scope at the base and let it be inherited by all subclasses. However, we can also exclude all "Object" instance_methods from the considered list. Both `Object.instance_methods` and `Object.private_instance_methods` have a lot of methods.

Reply:
Good call! Updated to dynamically get the files.

As for filtering out `Object.methods` or `Object.instance_methods`, it doesn't move the needle too much and could potentially cause issues if people define named scopes that collide so probably best to avoid for now.